### PR TITLE
Implement renderTemplate

### PR DESCRIPTION
**Pull request by TheOneRing**
The function alows to fill template files

### CODE REVIEW

**Review by fmoc — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/12051/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/12051/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/12051/guiReportUpload/serverlog.log

Screenshots:
- [Delete_a_folder.png](https://cache.owncloud.com/public/owncloud/client/12051/guiReportUpload/screenshots/Delete_a_folder.png)

</details>

File: src/common/utility.cpp [hunk 1]
```diff
@@ -606,4 +607,27 @@ QDebug &operator<<(QDebug &debug, nanoseconds in)
                  << ms.count() << "ms)";
 }
 
+QString Utility::renderTemplate(const QString &in, const QMap<QString, QString> &values)
+{
+    static const QRegularExpression pattern(QStringLiteral("@{([^{}]+)}"));
+    QString out = in;
+    const auto replace = [&out, &values](QRegularExpressionMatchIterator it) {
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/common/utility.cpp:614]: I found the while loop in the do{}while loop confusing...

File: src/gui/newwizard/setupwizardwindow.cpp [hunk 2]
```diff
@@ -17,12 +17,7 @@ using namespace OCC;
 
 QString replaceCssColors(QString stylesheet)
 {
-    QString rv = stylesheet;
-
-    rv = stylesheet.replace(QStringLiteral("@WIZARD_BACKGROUND_COLOR@"), Theme::instance()->wizardHeaderBackgroundColor().name());
-    rv = stylesheet.replace(QStringLiteral("@WIZARD_FONT_COLOR@"), Theme::instance()->wizardHeaderTitleColor().name());
-
-    return rv;
+    return Utility::renderTemplate(stylesheet, { { QStringLiteral("WIZARD_BACKGROUND_COLOR"), Theme::instance()->wizardHeaderBackgroundColor().name() }, { QStringLiteral("WIZARD_FONT_COLOR"), Theme::instance()->wizardHeaderTitleColor().name() } });
```
[the remark this answers is not in the document]
Comment (TheOneRing) [reply src/gui/newwizard/setupwizardwindow.cpp:20]: I formated it diffrently... maybe we can tweak clang format

[the remark this answers is not in the document]
Comment (fmoc) [reply src/gui/newwizard/setupwizardwindow.cpp]: Good idea. I also have some suggestions how to improve its behavior.

**Pull request merged**